### DISCUSSION TITLE
What are the default username and password emqx dashboard?

### DISCUSSION THREAD

**Discussion by byteshiva** [emqx/emqx#6901] [Q&A] [answered]
What are the default username and password emqx dashboard?

```
docker run -d --name emqx -p 1883:1883 -p 8083:8083 -p 8883:8883 -p 8084:8084 -p 18083:18083 emqx/emqx
```

![image](https://user-images.githubusercontent.com/10637299/151691640-dc4aab9c-8c4e-4290-9907-f62e046d1c27.png)

**Comment by byteshiva** [comment]
I see it has `admin` as username, what is the default password for `admin` user

**Comment by byteshiva** [comment, marked as answer]
The default username and password


```
username: admin
password: public
```

![image](https://user-images.githubusercontent.com/10637299/151692845-c9d31b60-9218-4115-b2a1-ffe420d34009.png)

**Comment by tunasakar** [comment]
for docker-compose...

    environment:
      - EMQX_DASHBOARD__DEFAULT_USERNAME=${EMQX_DASHBOARD__DEFAULT_USERNAME}
      - EMQX_DASHBOARD__DEFAULT_PASSWORD=${EMQX_DASHBOARD__DEFAULT_PASSWORD}

**Reply by AmirMassou94** [reply]
what if I want to user secret?
I see that :
 ```  
 EMQX_DASHBOARD__DEFAULT_USERNAME_FILE: /run/secrets/emqx_dashboard_username
 EMQX_DASHBOARD__DEFAULT_PASSWORD_FILE: /run/secrets/emqx_dashboard_password
 ```
lead to this error:
```
 emqx           | [warning] unknown_env_vars: ["EMQX_DASHBOARD__DEFAULT_PASSWORD_FILE", "EMQX_DASHBOARD__DEFAULT_USERNAME_FILE"]
```

**Reply by zmstone** [reply]
Hi @byteshiva 

Starting from 5.9, you can do this:

```
EMQX_DASHBOARD__DEFAULT_PASSWORD=file:///path/to/password/file
```

Ref: https://github.com/emqx/emqx/pull/14638

e.g.

```
$ echo aaa > /tmp/pwd
$ docker run --rm -it -p 18083:18083 -v /tmp/pwd:/pwd -e EMQX_DASHBOARD__DEFAULT_PASSWORD=file:///pwd emqx/emqx:5.9.0 emqx console
```

then you can login with username `admin` and pwd `aaa`
